Objects/bytearrayobject.c

```diff
@@ -585,7 +590,7 @@ bytearray_setslice_linear(PyByteArrayObject *self,
         buf = PyByteArray_AS_STRING(self);
     }
     else if (growth > 0) {
-        if (Py_SIZE(self) > (Py_ssize_t)PY_SSIZE_T_MAX - growth) {
+        if (Py_SIZE(self) > PyByteArray_SIZE_MAX - growth) {
```

Comment:
not in this PR, but it might make sense to make a `bytearray_resize_grow_lock_held(self, growth)` which could replace many of these "range check" + "PyErr_NoMemory"